packages/core/src/internal/fetchOptions.ts

```diff
@@ -75,6 +76,15 @@ export const makeFetchURL = (
   return finalUrl;
 };
 
+/** Extract the query parameters from a URL, without using {@link URL} to support partial URLs */
+const extractParams = (url: string): [string, URLSearchParams] => {
```

Comment:
```suggestion
const extractParams = (url: string): readonly [string, URLSearchParams] => {
```

nit: Let's omit the TSDoc comment here, since it doesn't add much given the short length of the code, and isn't public

I'd also rename the function from `extractParams` to `splitOutSearchParams`, or something similar for clarity

The `readonly` addition is for a different comment, just to make sure that always works